### ISSUE TITLE
Omnicomplete not working with multiple folders

### ISSUE THREAD

**Issue by Levlop** [lervag/vimtex#3048]
### Description

I want to create a latex structure, where I have main.tex in one folder, and chapter specific .tex files in another folder. 

I want to be a able to use omnicomplete to in order to autocomplete \ref{ and \cite{, but that doesn't work when the .tex file in not in the same directory as main.tex

### Steps to reproduce

These are all my relevant files, 
When I am in main.tex or i create a .tex file in the Main folder, I can type \ref{ and \cite{ and press <c-x> and <c-o> and a list of all the labels or citations will be displayed. It does show all the lables from the files in the subject folder, so it recognizes them. 

When I do this in any file in the Subject folder, this does not work. 

when :verbose set omnifunc is run in a tex buffer it says:  
omnifunc=vimtex#complete#omnifunc 
Last set from ~/.vim/plugged/vimtex/autoload/vimtex/complete.vim line 20 

Here are all my files: 

This is my tree: 

```└── VIM
    ├── Main
    │   ├── bibliography.bib
    │   ├── main.aux
    │   ├── main.bbl
    │   ├── main.blg
    │   ├── main.fdb_latexmk
    │   ├── main.fls
    │   ├── main.log
    │   ├── main.pdf
    │   ├── main.synctex.gz
    │   ├── main.tex
    │   └── main.toc
    └── Subjects
        ├── Subject1
        │   ├── subjecta.tex
        │   └── subjectb.tex
        └── Subject2
            └── subjectc.tex.

``` 
My main.tex looks like this: 

```
\documentclass[12pt,openany]{book}

\begin{document}
\section*{Subjects}

\input{../Subjects/Subject1/subjecta.tex}
\input{../Subjects/Subject1/subjectb.tex}
\input{../Subjects/Subject2/subjectc.tex}

\cite{test}

\newpage

\bibliographystyle{plain} 
\bibliography{bibliography}

\end{document} 
```

My subjecta/b/c files look like this: 

```
\section{a}\label{a}
This is section a
```
My bibliography looks like this:

```
@book{test,
  author    = {Leslie Lamport},
  title     = {LaTeX: A Document Preparation System},
  edition   = {2nd},
  publisher = {Addison-Wesley},
  year      = {1994},
  address   = {Boston}
}
```
and lastly my /.vimrc file looks like this: 

```
set number
cd ~/VIM/ 
filetype plugin on
colorscheme delek
syntax on

call plug#begin()
Plug 'lervag/vimtex'
call plug#end()

"Vimtex Setup
syntax enable
let g:vimtex_view_method = 'zathura'
let g:vimtex_compiler_method = 'latexmk'

"Vimtex settings
let g:tex_flavor = 'latex'
set conceallevel =1
let g:tex_conceal='abdmg'
highlight Conceal ctermbg=NONE guibg=NONE
let g:vimtex_view_general_options = '--synctex-forward @line:@col:@tex @pdf'

"omnifunction
set omnifunc=syntaxcomplete#Complete
```

### Expected behavior

<c-x> and <c-o> to display a list of all labels and citation after \ref{ or \cite{ when editing in the Subject Folder. 

### Actual behavior

Message displayed: 
 -- Omni completion (^O^N^P) Pattern not found 

### Do you use a latexmkrc file?

No

### VimtexInfo

```yaml
System info:
  OS: Ubuntu 24.04.1 LTS
  Vim version: VIM 9.1 (1-16, 647, 678, 697)
  Has clientserver: false

VimTeX project: main                                                                                                      base: main.tex                                                                                                          root: /home/lev/VIM/Main                                                                                                tex: /home/lev/VIM/Main/main.tex                                                                                        main parser: current file verified                                                                                      document class: book                                                                                                    source files:                                                                                                             main.tex                                                                                                                ../Subjects/Subject1/subjecta.tex                                                                                       ../Subjects/Subject1/subjectb.tex                                                                                       ../Subjects/Subject2/subjectc.tex                                                                                     compiler: latexmk                                                                                                         engine: -pdf                                                                                                            options:                                                                                                                  -verbose                                                                                                                -file-line-error                                                                                                        -synctex=1                                                                                                              -interaction=nonstopmode                                                                                              callback: 1                                                                                                             continuous: 1                                                                                                           executable: latexmk                                                                                                   viewer: Zathura                                                                                                           xwin id: 0
  qf method: LaTeX logfile
```

**Comment by lervag** [comment]
This is because the structure you want is hard to parse automatically. I.e., for completion to work properly, VimTeX needs to understand where the main file is. When you put the included files in a directory that is not a child directory of the main file, the automatic parsing will fail.

This should be relatively well explain in `:help vimtex-multi-file`. This section also explains the various ways to help VimTeX know where to find the main file. In your case, I think the easiest solution is to use the TeX root directive, i.e. at the top of your included files, add:

```tex
%! TeX root = ../main.tex
```
